platform/security/pdp/security-pdp-xacmlrealm/src/main/java/ddf/security/pdp/xacml/processor/BalanaPdp.java

```diff
@@ -305,23 +317,25 @@ public void endElement(String uri, String localName, String qName)
         }
 
         DOMResult domResult;
-        ClassLoader tccl = Thread.currentThread().getContextClassLoader();
-        Thread.currentThread().setContextClassLoader(BalanaPdp.class.getClassLoader());
+        ClassLoader tccl = Thread.currentThread()
```

Comment:
I meant that this block of code is setting a context classloader, transforming xml, and then resetting the context classloader at the end. This is exactly the sort of work the Parser was built to handle.